cranelift-codegen/meta/src/cdsl/operands.rs

```diff
@@ -115,18 +115,55 @@ pub(crate) enum OperandKindFields {
 #[derive(Clone, Debug)]
 pub(crate) struct OperandKind {
     pub name: &'static str,
-
-    doc: Option<String>,
-
-    pub default_member: Option<&'static str>,
-
+    doc: Option<&'static str>,
+    default_member: Option<&'static str>,
     /// The camel-cased name of an operand kind is also the Rust type used to represent it.
     pub rust_type: String,
-
     pub fields: OperandKindFields,
 }
 
 impl OperandKind {
+    fn new(
+        name: &'static str,
+        doc: Option<&'static str>,
+        default_member: Option<&'static str>,
+        rust_type: Option<&'static str>,
+        fields: OperandKindFields,
+    ) -> Self {
+        // Compute the default rust_type value, if it wasn't provided.
+        let rust_type = match rust_type {
+            Some(rust_type) => rust_type.to_string(),
+            None => match &fields {
+                OperandKindFields::ImmEnum(_) | OperandKindFields::ImmValue => {
+                    format!("ir::immediates::{}", camel_case(name))
+                }
+                OperandKindFields::VariableArgs => "&[Value]".to_string(),
+                OperandKindFields::TypeVar(_) | OperandKindFields::EntityRef => {
+                    format!("ir::{}", camel_case(name))
+                }
+            },
+        };
+        Self {
+            name,
+            doc,
+            default_member,
+            rust_type,
+            fields,
+        }
+    }
+
+    /// Name of this OperandKind in the format's member field.
+    pub fn default_member(&self) -> Option<&'static str> {
+        if let Some(member) = &self.default_member {
+            return Some(member);
+        }
+        match &self.fields {
+            OperandKindFields::ImmEnum(_) | OperandKindFields::ImmValue => Some("imm"),
```

Comment:
I was wondering about this a few days ago: if we were to have an instruction with multiple immediates then there would be a compilation error of the generated code, right?

Reply:
If they all have the default format field name, yeah.